python/lsst/obs/pfs/isrTask.py

```diff
@@ -107,37 +135,92 @@ def runDataRef(self, sensorRef):
+                t_read = self.config.brokenRedShutter.t_read
+                t_stare = self.config.brokenRedShutter.t_stare
+
+                N = result.exposure.getHeight()
+                kern = np.zeros((N, N))
+                kern += t_wipe/N
+                kern += t_read/N*np.tri(N, k=-1)
+                np.fill_diagonal(kern, t_stare)
+                kern /= t_exp
+
+                np.fill_diagonal(kern, 1 + kern.diagonal()) # i.e. add 1 to the diagonal
+                ikern = np.linalg.inv(kern)
+
+                self.log.info("Correcting for broken red shutter analytically")
+                result.exposure.image.array[:] = ikern@result.exposure.image.array
+            else:
+                #
+                # If we're using the `r` arm subtract a neighbouring image if it's a bias
+                #
+                # If the source of photons is stable, this corrects for the extra
+                # light accumulated during CCD wipe and readout
+                #
+                # We search for the bias in a window around the visit v0 in order
+                #   v0 + 1, v0 - 1, v0 + 2, v0 - 2, ...
+                # (if brokenRedShutter.causalWindow is True, only use v0 + 1, v0 + 2, ...)
+                #
+                # Don't fix a bias; this is especially important as if we do, it'll lead
+                # to an attempt to correct the bias we're using for the shutter correction
+                #
+                # N.b. we can't rely on the sensorRef.dataId because we adjusted the visit number
+                # before possibly calling this routine recursively to fetch the desired bias
+                md = sensorRef.get("raw_md")
+                if md.get('DATA-TYP', "").upper() != "BIAS":
+                    dataId = dict(arm = sensorRef.dataId["arm"],
+                                  filter = sensorRef.dataId["arm"], # yes, this is necessary
+                                  spectrograph = sensorRef.dataId["spectrograph"])
+                    dataId0 = sensorRef.dataId
+
+                    for dv in sum([[i, -i] for i in range(1, self.config.brokenRedShutter.window + 1)], []):
+                        if self.config.brokenRedShutter.causalWindow and dv < 0:
+                            continue
+
+                        exp2 = None
+                        try:
+                            sensorRef.dataId = dataId
+                            dataId["visit"] = dataId0["visit"] + dv + 10000
+
+                            md = sensorRef.get("raw_md")
+                            if md.get('DATA-TYP', "").upper() == "BIAS":
+                                self.log.info("Reading visit %d for broken red shutter correction",
+                                              sensorRef.dataId["visit"])
+
+                                exp2 = self.runDataRef(sensorRef).exposure
+                        except NoResults as e:
+                            print("RHL", e)
```

Comment:
Remove debug statement

Reply:
Actually the + 10000 is part of this debug code, so I need to remove that too (and the format is wrong -- clearly
I added this check-the-exception code but got distracted before doing the checking). I'm glad you spotted this.